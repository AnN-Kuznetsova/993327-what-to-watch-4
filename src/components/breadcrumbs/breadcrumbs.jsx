import PropTypes from "prop-types";
import React from "react";
import {Link} from "react-router-dom";


export const Breadcrumbs = (props) => {
  const {breadcrambsList} = props;

  return (
    <nav className="breadcrumbs">
      <ul className="breadcrumbs__list">
        {breadcrambsList.map((item, index) => {
          return (
            <li key={item.title + index} className="breadcrumbs__item">
              <Link
                className="breadcrumbs__link"
                to={item.link}
                style={item.link ? {} : {pointerEvents: `none`}}
              >{item.title}</Link>
            </li>

          );
        })}
      </ul>
    </nav>
  );
};


Breadcrumbs.propTypes = {
  breadcrambsList: PropTypes.arrayOf(PropTypes.shape({
    link: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  })),
};
