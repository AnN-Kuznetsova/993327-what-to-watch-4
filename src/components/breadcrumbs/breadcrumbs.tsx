
import * as React from "react";
import {Link} from "react-router-dom";


interface Props {
  breadcrambsList?: {
    link: string,
    title: string,
  }[];
}


export const Breadcrumbs: React.FunctionComponent<Props> = (props: Props) => {
  const {breadcrambsList} = props;

  return (
    <nav className="breadcrumbs">
      <ul className="breadcrumbs__list">
        {breadcrambsList.map((item, index) => {
          return (
            <li key={item.title + index} className="breadcrumbs__item">
              <Link
                className="breadcrumbs__link"
                to={item.link}
                style={item.link ? {} : {pointerEvents: `none`}}
              >{item.title}</Link>
            </li>

          );
        })}
      </ul>
    </nav>
  );
};
