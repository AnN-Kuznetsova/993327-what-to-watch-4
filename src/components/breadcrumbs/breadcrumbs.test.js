import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import {Breadcrumbs} from "./breadcrumbs";
import {history} from "../../history";


const breadcrambsList = [
  {
    title: `title`,
    link: `/films/1`,
    onLinkClick: () => {},
  },
  {
    title: `Add review`,
    link: ``,
  }
];


describe(`Render Breadcrambs`, () => {
  it(`Breadcrambs should match with snapshot`, () => {
    const breadcrambsSnapshot = renderer.create(
        <Router history={history} >
          <Breadcrumbs breadcrambsList={breadcrambsList} />
        </Router>
    ).toJSON();

    expect(breadcrambsSnapshot).toMatchSnapshot();
  });
});
