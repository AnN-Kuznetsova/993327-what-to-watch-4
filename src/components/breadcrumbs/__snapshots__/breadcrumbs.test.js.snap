// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Breadcrambs Breadcrambs should match with snapshot 1`] = `
<nav
  className="breadcrumbs"
>
  <ul
    className="breadcrumbs__list"
  >
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
        href="/films/1"
        onClick={[Function]}
        style={Object {}}
      >
        title
      </a>
    </li>
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        Add review
      </a>
    </li>
  </ul>
</nav>
`;
